src/include/add-on/add-on-workflow.rb

```diff
@@ -1175,32 +1121,29 @@ def RunAddOnMainDialog(enable_back, enable_next, enable_abort, back_button, next
 
           if ret2 == :next
             # Add-On product has been added, integrate it (change workflow, use y2update)
-            AddOnProduct.Integrate(AddOnProduct.src_id)
+            @added_repos.each { |src_id| AddOnProduct.Integrate(src_id) }
 
             # check whether it requests registration (FATE #301312)
-            AddOnProduct.PrepareForRegistration(AddOnProduct.src_id)
+            @added_repos.each { |src_id| AddOnProduct.PrepareForRegistration(src_id) }
             some_addon_changed = true
             # do not keep first_time, otherwise summary won't be shown during installation
             ret = nil if ret == :skip_to_add
           elsif ret2 == :abort || ret2 == :cancel
             Builtins.y2milestone("Add-on sequence aborted")
 
-            if AddOnProduct.src_id != nil
-              Builtins.y2milestone(
-                "Removing add-on repository: %1",
-                AddOnProduct.src_id
-              )
+            if !@added_repos.empty?
```

Comment:
I think it is not needed to check it. each will work fine with empty array and log that it removing empty repositories looks consistent for me.